src/lib/PnP.Framework/AuthenticationManager.cs

```diff
@@ -1061,13 +1064,33 @@ internal void ConfigureOnPremisesContext(string siteUrl, ClientContext clientCon
             clientContext.AddContextSettings(clientContextSettings);
         }
 
+        /// <summary>
+        /// Returns an app only AuthenticationManager object
+        /// </summary>
+        /// <param name="siteUrl">Site for which the ClientContext object will be instantiated</param>
+        /// <param name="realm">Realm of the environment (tenant) that requests the ClientContext object, may be null</param>
+        /// <param name="appId">Application ID which is requesting the ClientContext object</param>
+        /// <param name="appSecret">Application secret of the Application which is requesting the ClientContext object</param>
+        /// <param name="acsHostUrl">Azure ACS host, defaults to accesscontrol.windows.net but internal pre-production environments use other hosts</param>
+        /// <param name="globalEndPointPrefix">Azure ACS endpoint prefix, defaults to accounts but internal pre-production environments use other prefixes</param>
+        /// <returns>AuthenticationManager that can be used to create authenticated ClientContext</returns>
+        public static async Task<AuthenticationManager> CreateACSAppOnlyAsync(string siteUrl, string realm, string appId, string appSecret, string acsHostUrl = "accesscontrol.windows.net", string globalEndPointPrefix = "accounts")
+        {
+            var acsTokenProvider = await ACSTokenGenerator.GetACSAuthenticationProviderAsync(new Uri(siteUrl), realm, appId, appSecret, acsHostUrl, globalEndPointPrefix);
+            return new AuthenticationManager(acsTokenProvider);
+        }
+
+        const string GetACSAppOnlyContext_ObsoletionMessage =
+            "Use AuthenticationManager.CreateACSAppOnlyAsync() and GetContextAsync()";
+
         /// <summary>
         /// Returns an app only ClientContext object
         /// </summary>
         /// <param name="siteUrl">Site for which the ClientContext object will be instantiated</param>
         /// <param name="appId">Application ID which is requesting the ClientContext object</param>
         /// <param name="appSecret">Application secret of the Application which is requesting the ClientContext object</param>
         /// <returns>ClientContext to be used by CSOM code</returns>
+        [Obsolete(GetACSAppOnlyContext_ObsoletionMessage)]

```

Comment:
@fowl2 : why are you marking the ACS methods as obsolete?

Reply:
because they're inconsistent with the other authentication methods, which complicates token caching